src/modules/uniplayer/Uniplayer.jsx

```diff
@@ -0,0 +1,189 @@
+import './uniplayer.scss';
+
+import React from 'react';
+import ReactPlayer from 'react-player';
+import Duration from './Duration.jsx';
+
+export default class Uniplayer extends React.Component {
+
+  constructor(props) {
+   super(props);
+
+   this.state = {
+
+     //Player State
+     playing: true,
+     volume: 1,
+     muted: false,
+     played: 0,
+     loaded: 0,
+     duration: 0,
+     playbackRate: 1.0,
+     loop: false,
+     seeking: false,
+   };
+
+   //React Player Functions
+   this.stop = this.stop.bind(this);
+   this.toggleLoop = this.toggleLoop.bind(this);
+   this.setVolume = this.setVolume.bind(this);
+   this.toggleMuted = this.toggleMuted.bind(this);
+   this.setPlaybackRate = this.setPlaybackRate.bind(this);
+   this.onPlay = this.onPlay.bind(this);
+   this.onPause = this.onPause.bind(this);
+   this.onSeekMouseDown = this.onSeekMouseDown.bind(this);
+   this.onSeekChange = this.onSeekChange.bind(this);
+   this.onSeekMouseUp = this.onSeekMouseUp.bind(this);
+   this.onProgress = this.onProgress.bind(this);
+   this.onEnded = this.onEnded.bind(this);
+   this.onDuration = this.onDuration.bind(this);
+
+   //OpenRecord Player Addon Functions
+   this.playToggle = this.playToggle.bind(this);
+ }
+
+ //React Player Functions
+ stop(){
+   this.setState({ url: null, playing: false })
+ }
+ toggleLoop(){
+   this.setState({ loop: !this.state.loop })
+ }
+ setVolume(e){
+   this.setState({ volume: parseFloat(e.target.value) })
+ }
+ toggleMuted(){
+   this.setState({ muted: !this.state.muted })
+ }
+ setPlaybackRate(e){
+   this.setState({ playbackRate: parseFloat(e.target.value) })
+ }
+ onPlay(){
+   this.setState({ playing: true })
+ }
+ onPause(){
+   this.setState({ playing: false })
+ }
+ onEnded(){
+   if(this.state.repeat){
+     if(this.state.isYouTube){
+       this.YTPlayer.seekTo(0);
+     } else {
+       this.SCPlayer.seekTo(0);
+     }
```

Comment:
It's a good idea in general to avoid nested if-statements:
```javascript
if (this.state.repeat && this.isYouTube) {
    this.YTPlayer.seekTo(0);
} else if (this.state.repeat) {
    this.SCPlayer.seekTo(0);
} else {
    this.autoplayNext();
}
```